airbyte-db/db-lib/src/main/java/io/airbyte/db/instance/jobs/migrations/V0_40_18_001__AddProgressBarStats.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.db.instance.jobs.migrations;
+
+import static org.jooq.impl.DSL.currentOffsetDateTime;
+import static org.jooq.impl.DSL.field;
+import static org.jooq.impl.DSL.foreignKey;
+import static org.jooq.impl.DSL.primaryKey;
+import static org.jooq.impl.DSL.unique;
+
+import java.time.OffsetDateTime;
+import java.util.UUID;
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.jooq.DSLContext;
+import org.jooq.Field;
+import org.jooq.impl.DSL;
+import org.jooq.impl.SQLDataType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The estimated columns contains the overall estimated records and bytes for an attempt.
+ * <p>
+ * The new stream_stats table contains the estimated and emitted records/bytes for an attempt at the
+ * per-stream level. This lets us track per-stream stats as an attempt is in progress.
+ */
+public class V0_40_18_001__AddProgressBarStats extends BaseJavaMigration {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(V0_40_18_001__AddProgressBarStats.class);
+
+  @Override
+  public void migrate(final Context context) throws Exception {
+    LOGGER.info("Running migration: {}", this.getClass().getSimpleName());
+
+    // Warning: please do not use any jOOQ generated code to write a migration.
+    // As database schema changes, the generated jOOQ code can be deprecated. So
+    // old migration may not compile if there is any generated code.
+    try (final DSLContext ctx = DSL.using(context.getConnection())) {
+      addEstimatedColumnsToSyncStats(ctx);
+      addStreamStatsTable(ctx);
+    }
+  }
+
+  private static void addEstimatedColumnsToSyncStats(final DSLContext ctx) {
+    ctx.alterTable("sync_stats")
+        .add(
+            field("estimated_records", SQLDataType.BIGINT.nullable(true)),
+            field("estimated_bytes", SQLDataType.BIGINT.nullable(true)))
+        .execute();
+  }
+
+  private static void addStreamStatsTable(final DSLContext ctx) {
+    // Metadata Columns
+    final Field<UUID> id = field("id", SQLDataType.UUID.nullable(false));
+    final Field<Integer> attemptId = field("attempt_id", SQLDataType.INTEGER.nullable(false));
+    final Field<String> streamName = field("stream_name", SQLDataType.VARCHAR.nullable(false));
```

Comment:
We should add `stream_namespace` as well since `stream_name` could collision.

Reply:
good point!

@evantahler stream_namespace is something we might need to change on the FE too.